src/enumerate.js

```diff
@@ -0,0 +1,188 @@
+const js2xmlparser = require('js2xmlparser');
+let winrm_soap_req = require('./base-request.js');
+let winrm_http_req = require('./http.js');
+
+function constructBeginEnumerationRequest(_params) {
+    var res = winrm_soap_req.getSoapHeaderRequest({
+        'resource_uri': _params.resourceUri || 'http://schemas.dmtf.org/wbem/wscim/1/*',
+        'action': 'http://schemas.xmlsoap.org/ws/2004/09/enumeration/Enumerate',
+    });
+
+    res['s:Body'] = {
+        'wsen:Enumerate': [{
+            'wsen:NewContext': {},
+            'wsman:OptimizeEnumeration': [{}],
+            'wsen:MaxTime': _params.maxTime || 'PT60S',
+            'wsman:MaxElements': _params.maxElements || 20
+        }]
+    };
+
+    if (_params.filter) {
+        res['s:Body']['wsen:Enumerate'][0]['wsman:Filter'] = [{
+            '@': {
+              'Dialect': _params.filterDialect || 'http://www.w3.org/TR/1999/REC-xpath-19991116'
+            },
+            '#':  _params.filter
+        }];
+    }
+
+    return js2xmlparser.parse('s:Envelope', res);
+}
+
+function constructPullEnumerationRequest(_params) {
+    var res = winrm_soap_req.getSoapHeaderRequest({
+        'resource_uri': _params.resourceUri || 'http://schemas.dmtf.org/wbem/wscim/1/*',
+        'action': 'http://schemas.xmlsoap.org/ws/2004/09/enumeration/Pull'
+    });
+
+    res['s:Body'] = {
+        'wsen:Pull': [{
+            'wsen:EnumerationContext': _params.enumerationId,
+            'wsen:MaxTime': _params.maxTime || 'PT10S',
+            'wsen:MaxElements': _params.maxElements || 20
+        }]
+    };
+
+    return js2xmlparser.parse('s:Envelope', res);
+}
+
+function constructReleaseEnumerationRequest(_params) {
+    var res = winrm_soap_req.getSoapHeaderRequest({
+        'resource_uri': _params.resourceUri || 'http://schemas.dmtf.org/wbem/wscim/1/*',
+        'action': 'http://schemas.xmlsoap.org/ws/2004/09/enumeration/Release'
+    });
+
+    res['s:Body'] = {
+        'wsen:Release': [{
+            'wsen:EnumerationContext': _params.enumerationId
+        }]
+    };
+
+    return js2xmlparser.parse('s:Envelope', res);
+}
+
+function getObjects(items) {
+    let itemObjects = [];
+    if (items) {
+        for (let item of items[0][Object.keys(items[0])[0]]) {
+            let itemObject = {};
+            // TODO return map of multiple item keys?
+            Object.keys(item).forEach((k) => {
```

Comment:
why mix `for` and `forEach` here - nicer to stick to one or the other for both loops?